src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs

```diff
@@ -998,5 +983,84 @@ public TestAccessor(TextBox textBox)
 
             public bool ShouldRenderPlaceHolderText(in Message m) => _textBox.ShouldRenderPlaceHolderText(m);
         }
+
+        protected override AccessibleObject CreateAccessibilityInstance()
+           => new TextBoxAccessibleObject(this);
+
+        [ComVisible(true)]
+        internal class TextBoxAccessibleObject : ControlAccessibleObject
+        {
+            internal TextBoxAccessibleObject(TextBox owner) : base(owner)
+            {
+            }
+
+            internal new string TextLabel
+            {
+                get
+                {
+                    // Otherwise use the text of the preceding Label control, if there is one
+                    Label previousLabel = PreviousLabel;
+                    if (previousLabel != null)
+                    {
+                        string text = previousLabel.Text;
+                        if (!string.IsNullOrEmpty(text))
+                        {
+                            return text;
+                        }
+                    }
+
+                    // This control has no discernable MSAA name - return an empty string to indiciate 'nameless'.
+                    return null;
+                }
+            }
+
+            public override string Name
+            {
+                get
+                {
+                    string name = WindowsFormsUtils.TextWithoutMnemonics(TextLabel);
+                    if (name == null)
+                    {
+                        name = "";
+                    }
+
+                    // Otherwise just return the default label string, minus any mnemonics
+                    return name;
+                }
+                set => base.Name = value;
+            }
+
+            private TextBox OwningTextBox => Owner as TextBox;
+
+            internal override bool IsIAccessibleExSupported() => true;
```

Comment:
Not sure that we need this. Maybe we can set it to true in base `ControlAccessibleObject` as well as `SupportsUiaProviders`.